source/patterns/@aws-solutions-constructs/aws-lambda-opensearch/README.md

```diff
@@ -0,0 +1,164 @@
+import * as lambda from "aws-cdk-lib/aws-lambda";
+
+const lambdaProps: lambda.FunctionProps = {
+  code: lambda.Code.fromAsset(`lambda`),
+  runtime: lambda.Runtime.NODEJS_14_X,
+  handler: 'index.handler'
+};
+
+new LambdaToOpenSearch(this, 'sample', {
+  lambdaFunctionProps: lambdaProps,
+  domainName: 'testdomain',
+  // TODO: Ensure the Cognito domain name is globally unique
+  cognitoDomainName: 'globallyuniquedomain' + Aws.ACCOUNT_ID
+});
+```
+
+Python
+```python
+from aws_solutions_constructs.aws_lambda_opensearch import LambdaToOpenSearch
+from aws_cdk import (
+    aws_lambda as _lambda,
+    Aws,
+    Stack
+)
+from constructs import Construct
+
+lambda_props = _lambda.FunctionProps(
+    code=_lambda.Code.from_asset('lambda'),
+    runtime=_lambda.Runtime.PYTHON_3_9,
+    handler='index.handler'
+)
+
+LambdaToOpenSearch(self, 'sample',
+                            lambda_function_props=lambda_props,
+                            domain_name='testdomain',
+                            # TODO: Ensure the Cognito domain name is globally unique
+                            cognito_domain_name='globallyuniquedomain' + Aws.ACCOUNT_ID
+                            )
+```
+
+Java
+``` java
+import software.constructs.Construct;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.Aws;
+import software.amazon.awscdk.services.lambda.*;
+import software.amazon.awscdk.services.lambda.Runtime;
+import software.amazon.awsconstructs.services.lambdaopensearch.*;
+
+new LambdaToOpenSearch(this, "sample",
+        new LambdaToOpenSearchProps.Builder()
+                .lambdaFunctionProps(new FunctionProps.Builder()
+                        .runtime(Runtime.NODEJS_14_X)
+                        .code(Code.fromAsset("lambda"))
+                        .handler("index.handler")
+                        .build())
+                .domainName("testdomain")
+                // TODO: Ensure the Cognito domain name is globally unique
+                .cognitoDomainName("globallyuniquedomain" + Aws.ACCOUNT_ID)
+                .build());
+```
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingLambdaObj?|[`lambda.Function`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html)|Existing instance of Lambda Function object, providing both this and `lambdaFunctionProps` will cause an error.|
+|lambdaFunctionProps?|[`lambda.FunctionProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionProps.html)|User provided props to override the default props for the Lambda function.|
+|domainProps?|[`opensearchservice.CfnDomainProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_opensearchservice.CfnDomainProps.html)|Optional user provided props to override the default props for the OpenSearch Service.|
```

Comment:
I like this better than osDomainProps, but since there are multiple domains being discussed I think these need to be:

openSearchDomainProps
openSearchDomainName (and the docs just say it's used for OpenSearch)
cognitoDomainName (and the docs say if it defaults to the openSearchDomainName)